WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1363,25 +1361,57 @@ public void onImageSettingsRequested(EditorImageMetaData editorImageMetaData) {
         MediaSettingsActivity.showForResult(this, mSite, editorImageMetaData);
     }
 
-    @Override public void onPositiveClicked(String instanceTag) {
-        if (instanceTag != null) {
-            if (instanceTag == TAG_PUBLISH_CONFIRMATION_DIALOG) {
+    @Override public void onPositiveClicked(@NonNull String instanceTag) {
+        switch (instanceTag) {
+            case TAG_PUBLISH_CONFIRMATION_DIALOG:
                 mPost.setStatus(PostStatus.PUBLISHED.toString());
                 publishPost();
-            } else if (instanceTag == TAG_REMOVE_FAILED_UPLOADS_DIALOG) {
+                break;
+            case TAG_REMOVE_FAILED_UPLOADS_DIALOG:
                 // Clear failed uploads
                 mEditorFragment.removeAllFailedMediaUploads();
-            }
-        } else {
-            ToastUtils.showToast(EditPostActivity.this,
-                    getString(R.string.editor_toast_no_action_taken));
-            AppLog.e(T.EDITOR, "Dialog instanceTag is null - positive button clicked, but no action taken");
+                break;
+            case ASYNC_PROMO_DIALOG_TAG:
+                publishPost();
+                break;
+            default:
+                AppLog.e(T.EDITOR, "Dialog instanceTag is not recognized");
+                throw new UnsupportedOperationException("Dialog instanceTag is not recognized");
         }
     }
 
-    @Override public void onNegativeClicked(String instanceTag) {
-        if (instanceTag != null && instanceTag == TAG_PUBLISH_CONFIRMATION_DIALOG) {
-            // no op
+    @Override public void onNegativeClicked(@NonNull String instanceTag) {
+        switch (instanceTag) {
+            case ASYNC_PROMO_DIALOG_TAG:
+                PromoDialog promoDialog = (PromoDialog) getSupportFragmentManager().findFragmentByTag(instanceTag);
+                if (promoDialog != null) {
+                    promoDialog.dismiss();
+                }
+                break;
+            case TAG_PUBLISH_CONFIRMATION_DIALOG:
```

Comment:
Do we really need to manually dismiss the Alert dialog? AFAIK an Alert dialog is automatically dismissed after the 'OnClickListener' is invoked ([see](https://android.googlesource.com/platform/frameworks/base.git/+/master/core/java/com/android/internal/app/AlertController.java) line 144 ). WDYT?

Reply:
That's true for regular dialogs with default buttons. But not when you use custom content view with your own buttons.